templates/editConn.js

```diff
@@ -80,19 +69,35 @@ function getData() {
 function validTextInput(id, errMsg, regex, regexErr) {
   let wrapper = document.getElementById(id);
   let input = wrapper.getElementsByTagName("input")[0];
-  wrapper.classList.remove("error");
+  clearErr(wrapper);
   let v = input.value.trim();
   if (v === "") {
-    wrapper.classList.add("error");
-    err.push(errMsg);
+    setErr(wrapper, errMsg);
     return "";
   }
   if (regex && regexErr) {
     if (!v.match(regex)) {
-      wrapper.classList.add("error");
-      err.push(regexErr);
+      setErr(wrapper, regexErr);
       return "";
     }
   }
   return v;
 }
+
+function clearErr(wrapper) {
+  wrapper.classList.remove("error");
+  let ele = wrapper.getElementsByClassName("errMsg")[0];
```

Comment:
We might want to check this element is found

Reply:
I think it's safe to without additional check. Failures like this one will block the message being sent, so it's easier for developer to figure out where the nil pointer started.